sopel/bot.py

```diff
@@ -659,38 +663,73 @@ def dispatch(self, pretrigger):
         for priority in ('high', 'medium', 'low'):
             items = self.get_triggered_callables(priority, pretrigger, blocked)
             for func, trigger, is_blocked in items:
+                function_name = "%s.%s" % (func.__module__, func.__name__)
                 # skip running blocked functions, but track them for logging
                 if is_blocked:
-                    function_name = "%s.%s" % (
-                        func.__module__, func.__name__
-                    )
                     list_of_blocked_functions.append(function_name)
                     continue
 
                 # call triggered function
                 wrapper = SopelWrapper(
                     self, trigger, output_prefix=func.output_prefix)
                 if func.thread:
+                    # run in a separate thread
                     targs = (func, wrapper, trigger)
                     t = threading.Thread(target=self.call, args=targs)
+                    t.name = '%s-%s' % (t.name, function_name)
```

Comment:
Just a quick note on that line: this will be of tremendous help when debugging threads, because now we'll be able to print the current thread's name, and it will contain the function triggered!

Or even, if someone is fond of debug tools/memory inspection/threading inspection, they'll be able to know exactly what's going on at any given time.

This is just the little thing that help with observability.